framework/src/kernels/Kernel.C

```diff
@@ -94,8 +94,14 @@ Kernel::computeResidual()
 
   precalculateResidual();
   for (_i = 0; _i < _test.size(); _i++)
+  {
     for (_qp = 0; _qp < _qrule->n_points(); _qp++)
+    {
+      if (_local_ref.size())
+        _local_ref(_i) += std::abs(_JxW[_qp] * _coord[_qp] * computeQpResidual());
       _local_re(_i) += _JxW[_qp] * _coord[_qp] * computeQpResidual();
```

Comment:
@dschwen would it be define a new local variable, e.g.:
```
const auto res = _JxW[_qp] * _coord[_qp] * computeQpResidual();
```

Reply:
I think so. You should add it if it s not in there.

Reply:
The compiler might be smart enough to optimize this, but it won't hurt to add the local variable.